ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/websocket/DefaultWebSocketSessionImpl.kt

```diff
@@ -131,6 +131,9 @@ class DefaultWebSocketSessionImpl(
             ponger.close()
             last?.release()
             filtered.close()
+
+            // Job will be cancelled if it's not completed yet.
+            closeReasonRef.cancel()
```

Comment:
Because of this change we will never have a real close reason received from a peer, always cancellation. It will only have a value if we close the session first but not when a peer initiates the termination.

Reply:
Otherwise, we need a test to ensure that.